src/entities/ChannelMetadata.ts

```diff
@@ -0,0 +1,23 @@
+import { Subscription } from './Subscription';
+import { SubscriptionOptions } from './commonTypes';
+
+export class ChannelMetadata {
+  private id: string;
+  private eventEmitter: any;
+  private pubnub: any;
+
+  constructor(id: string, eventEmitter: any, pubnub: any) {
+    this.id = id;
+    this.eventEmitter = eventEmitter;
+    this.pubnub = pubnub;
+  }
+  subscription(subscriptionOptions?: SubscriptionOptions) {
+    return new Subscription({
+      channels: subscriptionOptions?.receivePresenceEvents ? [this.id, `${this.id}-pnpres`] : [this.id],
```

Comment:
I believe we agreed that there is no point in subscription for App Context object presence channels?

Reply:
Yes, missed.

Reply:
updated!